server/api/plextv.ts

```diff
@@ -87,6 +126,73 @@ class PlexTvAPI {
     });
   }
 
+  public async getDevices(): Promise<PlexDevice[]> {
+    try {
+      const devicesResp = await this.axios.get(
+        '/api/resources?includeHttps=1',
+        {
+          transformResponse: [],
+          responseType: 'text',
+        }
+      );
+      const parsedXml = await xml2js.parseStringPromise(
+        devicesResp.data as DeviceResponse
+      );
+      const devResp = (): PlexDevice[] => {
+        const arr: PlexDevice[] = [];
+        parsedXml?.MediaContainer?.Device?.forEach((pxml: DeviceResponse) => {
+          const connArr: PlexConnection[] = [];
+          pxml?.Connection?.forEach((conn: ConnectionResponse) => {
+            connArr.push({
+              protocol: conn.$.protocol,
+              address: conn.$.address,
+              port: parseInt(conn.$.port, 10),
+              uri: conn.$.uri,
+              local: conn.$.local == '1' ? true : false,
+            });
+          });
+          arr.push({
+            name: pxml.$.name,
+            product: pxml.$.product,
+            productVersion: pxml.$.productVersion,
+            platform: pxml.$?.platform,
+            platformVersion: pxml.$?.platformVersion,
+            device: pxml.$?.device,
+            clientIdentifier: pxml.$.clientIdentifier,
+            createdAt: new Date(parseInt(pxml.$?.createdAt, 10) * 1000),
+            lastSeenAt: new Date(parseInt(pxml.$?.lastSeenAt, 10) * 1000),
+            provides: pxml.$.provides.split(','),
+            owned: pxml.$.owned == '1' ? true : false,
+            accessToken: pxml.$?.accessToken,
+            publicAddress: pxml.$?.publicAddress,
+            publicAddressMatches:
+              pxml.$?.publicAddressMatches == '1' ? true : false,
+            httpsRequired: pxml.$?.httpsRequired == '1' ? true : false,
+            synced: pxml.$?.synced == '1' ? true : false,
+            relay: pxml.$?.relay == '1' ? true : false,
+            dnsRebindingProtection:
+              pxml.$?.dnsRebindingProtection == '1' ? true : false,
+            natLoopbackSupported:
+              pxml.$?.natLoopbackSupported == '1' ? true : false,
+            presence: pxml.$?.presence == '1' ? true : false,
+            ownerID: pxml.$?.ownerID,
+            home: pxml.$?.home == '1' ? true : false,
+            sourceTitle: pxml.$?.sourceTitle,
+            connection: connArr,
+          });
+        });
+        return arr;
+      };
+      return devResp();
+    } catch (e) {
+      logger.error(
+        `Something went wrong getting the devices from plex.tv: ${e.message}`,
+        { label: 'Plex.tv API' }
```

Comment:
I know this is a bit mixed and matched around the app but trying to keep these the same going forward. Instead of the error message in the template literal, add it as a property `errorMessage`:

```suggestion
        'Something went wrong getting the devices from plex.tv',
        { label: 'Plex.tv API', errorMessage: e.message }
```